src/NextGen/DonationForm/Actions/ConvertDonationFormBlocksToFieldsApi.php

```diff
@@ -146,6 +148,73 @@ protected function createNodeFromDonorNameBlock(BlockModel $block): Node
         });
     }
 
+    /**
+     * @unreleased
+     */
+    protected function createNodeFromAmountBlock(BlockModel $block): Node
+    {
+        return DonationAmount::make('donationAmount')->tap(function (Group $group) use ($block) {
+            $amountRules = ['required', 'numeric'];
+
+            if ($block->hasAttribute('customAmount') && $block->getAttribute('customAmount') === true) {
+                if ($block->hasAttribute('customAmountMin')) {
+                    $amountRules[] = "min:{$block->getAttribute('customAmountMin')}";
+                }
+
+                if ($block->hasAttribute('customAmountMax') && $block->getAttribute('customAmountMax') > 0) {
+                    $amountRules[] = "max:{$block->getAttribute('customAmountMax')}";
+                }
+            }
```

Comment:
@jonwaldstein If the donation is a fixed amount we should be doing `size:$amount` to validate that it doesn't change.